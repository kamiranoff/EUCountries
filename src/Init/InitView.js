import React from 'react';
import { PropTypes } from 'prop-types';
import { StyleSheet, Text, TouchableOpacity, View, Dimensions } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000',
    alignItems: 'center',
    justifyContent: 'center',
  },
  countries: {
    color: 'white',
    fontSize: 40,
    fontWeight: '900',
    textAlign: 'center',
  },
  buttonContainer: {
    position: 'absolute',
    bottom: '10%',
  },
  button: {
    backgroundColor: 'blue',
    padding: 20,
    width: Dimensions.get('window').width * 0.9,
    alignItems: 'center',
  },
  buttonText: {
    color: 'white',
    fontSize: 25,
    fontWeight: '700',
  },
  error: {
    color: 'red',
    textAlign: 'center',
  },
  errorButton: {
    opacity: 0.5,
  },
});
const InitView = ({ handlePress, error }) => (
  <View style={styles.container}>
    <Text style={styles.countries}>EU</Text>
    <Text style={styles.countries}>COUNTRIES</Text>
    {error &&
    <Text style={styles.error}>Could not download data. Please make sure you are connected to the internet</Text>}
    <TouchableOpacity style={styles.buttonContainer} onPress={handlePress} disabled={!!error}>
      <View style={[styles.button, !!error && styles.errorButton]}>
        <Text style={styles.buttonText}>START</Text>
      </View>
    </TouchableOpacity>
  </View>
);

InitView.propTypes = {
  handlePress: PropTypes.func.isRequired,
  error: PropTypes.string,
};

export default InitView;
