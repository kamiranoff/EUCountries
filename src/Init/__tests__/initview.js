import React from 'react';
import renderer from 'react-test-renderer';
import InitView from '../InitView';

describe('InitView', () => {
  it('should render the component', () => {
    const wrapper = renderer.create(<InitView handlePress={() => null} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
})