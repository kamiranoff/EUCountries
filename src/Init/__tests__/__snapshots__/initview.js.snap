// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitView should render the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 40,
        "fontWeight": "900",
        "textAlign": "center",
      }
    }
  >
    EU
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 40,
        "fontWeight": "900",
        "textAlign": "center",
      }
    }
  >
    COUNTRIES
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": "10%",
        "opacity": 1,
        "position": "absolute",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "blue",
            "padding": 20,
            "width": 675,
          },
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 25,
            "fontWeight": "700",
          }
        }
      >
        START
      </Text>
    </View>
  </View>
</View>
`;
